src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodec.java

```diff
@@ -0,0 +1,376 @@
+    }
+
+    // ============================================================================================================
+    // Trivial override methods that are pulled form AsciiFeatureCodec
+    // This was done to ensure that this was a reasonable Codec class (with good interfaces for reading features).
+
+    @Override
+    public void close(final LineIterator lineIterator) {
+        CloserUtil.close(lineIterator);
+    }
+
+    @Override
+    public boolean isDone(final LineIterator lineIterator) {
+        return !lineIterator.hasNext();
+    }
+
+    @Override
+    public LineIterator makeSourceFromStream(final InputStream bufferedInputStream) {
+        return new LineIteratorImpl(new SynchronousLineReader(bufferedInputStream));
+    }
+
+    @Override
+    public FeatureCodecHeader readHeader(final LineIterator lineIterator) throws IOException {
+        return new FeatureCodecHeader(readActualHeader(lineIterator), FeatureCodecHeader.NO_HEADER_END);
+    }
+
+    @Override
+    public LocationAware makeIndexableSourceFromStream(final InputStream bufferedInputStream) {
+        final PositionalBufferedStream pbs;
+        if (bufferedInputStream instanceof PositionalBufferedStream) {
+            pbs = (PositionalBufferedStream) bufferedInputStream;
+        } else {
+            pbs = new PositionalBufferedStream(bufferedInputStream);
+        }
+        return new AsciiLineReaderIterator(new AsciiLineReader(pbs));
+    }
+
+    // ============================================================================================================
+
+    @Override
+    public GencodeGtfFeature decode(final LineIterator lineIterator) {
+
+        GencodeGtfFeature decodedFeature = null;
+
+        // Create some caches for our data (as we need to group it):
+        GencodeGtfGeneFeature gene = null;
+        GencodeGtfTranscriptFeature transcript = null;
+        final ArrayList< GencodeGtfExonFeature > exonStore = new ArrayList<>();
+        final ArrayList< GencodeGtfFeature > leafFeatureStore = new ArrayList<>();
+
+        // Accumulate lines until we have a full gene and all of its internal features:
+        while ( lineIterator.hasNext() ) {
+
+            String line = lineIterator.peek();
+
+            // We must assume we can get header lines.
+            // If we get a header line, we return null.
+            // This allows indexing to work.
+            if ( line.startsWith(COMMENT_START) ) {
+                lineIterator.next();
+                return null;
+            }
+
+            String[] splitLine = line.split("\t", -1);
+
+            // Ensure the file is at least trivially well-formed:
+            if (splitLine.length != NUM_COLUMNS) {
+                throw new UserException.MalformedFile("Found an invalid number of columns in the given GENCODE file on line "
+                        + currentLineNum + " - Given: " + splitLine.length + " Expected: " + NUM_COLUMNS + " : " + line);
```

Comment:
@droazen I don't think so.  There's no public interface in LineIterator to get the originating filename.